cloud/linode/instances.go

```diff
@@ -285,6 +285,13 @@ func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloud
 		addresses = append(addresses, v1.NodeAddress{Type: ip.ipType, Address: ip.ip})
 	}
 
+	// include IPs set by kubelet for internal node IP
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == v1.NodeInternalIP {
+			addresses = append(addresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: addr.Address})
+		}
+	}
```

Comment:
Should we make sure same ip is not already present in the slice? For example, if this InstanceMetadata() gets called again after 5-10 mins, then we'll end up adding all the existing ip's again into the slice. Not sure if k8s automatically handles such case.